core/src/services/box/core.rs

```diff
@@ -0,0 +1,377 @@
+
+                    if let Some(entry) = result
+                        .entries
+                        .iter()
+                        .find(|entry| entry.item_type == target_type && entry.name == target_name)
+                    {
+                        parent_id = entry.id.clone();
+                    } else {
+                        return Err(Error::new(ErrorKind::Unexpected, &format!("Can't find given folder, Please ensure that the folder path exists and is unique." )));
+                    }
+                }
+                _ => Err(parse_error(resp).await?),
+            }
+        }
+        let mut cache_guard = self.path_cache.lock().await;
+        cache_guard.insert(path, parent_id.clone());
+
+        Ok(parent_id)
+    }
+    pub async fn box_get(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let url: String = format!(
+            "https://api.box.com/2.0/files/{}/content",
+            self.get_file_id_by_path(path).await?
+        );
+        let mut request = Request::get(&url)
+            .header(CONTENT_TYPE, "application/json")
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+
+        self.sign(&mut request).await?;
+        self.client.send(request).await
+    }
+
+    pub async fn box_upload(
+        &self,
+        path: &str,
+        size: Option<usize>,
+        content_type: Option<&str>,
+        body: AsyncBody,
+    ) -> Result<Response<IncomingAsyncBody>> {
+        let folder_path=self.truncate_filename(path.trim_end_matches('/'));
+        let file_name="test.txt";
+        let url: String = "https://upload.box.com/api/2.0/files/content".to_string();
+        let mut req = Request::post(&url);
+        req = req.header(CONTENT_TYPE, "multipart/form-data; boundary=my-boundary");
+        let mut req_body = BytesMut::with_capacity(100);
+        write!(
+            &mut req_body,
+            "--my-boundary\ncontent-disposition: form-data; name=\"attributes\"\n\n{{\"name\":\"{}\", \"parent\":{{\"id\":\"{}\"}}}}\n--my-boundary\n",
+            file_name,
+            folder_path// should be folder id
+        ).unwrap();
+
+        write!(
+            &mut req_body,
+            "content-disposition: form-data; name=\"file\"; filename=\"{}\"\n",
+            file_name,
+        ).unwrap();
+        if let Some(mime) = content_type {
+            write!(&mut req_body, "Content-Type: {}\n\n", mime).unwrap();
+        } else {
+            write!(&mut req_body, "Content-Type: application/octet-stream\n\n").unwrap();
+        }
+        if let AsyncBody::Bytes(bytes) = body {
+            req_body.extend_from_slice(&bytes);
+        }
+        write!(&mut req_body, "\n--my-boundary").unwrap();
+        let req_body = AsyncBody::Bytes(req_body.freeze());
+        let mut request = req.body(req_body).map_err(new_request_build_error)?;
+
+
+        self.sign(&mut request).await?;
+        self.client.send(request).await
+    }
```

Comment:
the upload api reference: 
https://developer.box.com/guides/uploads/direct/file/ 
https://developer.box.com/reference/post-files-content/
the post message seems like 
<img width="619" alt="image" src="https://github.com/apache/incubator-opendal/assets/48410497/c9410076-e651-43f3-84bd-0865473d533d">
I wonder if it's correct to implement this api in this way

Reply:
I tried to use FormDataPart in  http_util/multpart.rs, It seems can fix the filename field in the post message, should we build the request body manually?

Reply:
if someone could provide me with some guidance, I would be very grateful.

Reply:
cc @Xuanwo